Source/JavaScriptCore/assembler/X86Assembler.h

```diff
@@ -2416,6 +2439,143 @@ class X86Assembler {
+        }
+    }
+
+    void vectorFsub(SIMDLane lane, XMMRegisterID left, XMMRegisterID right, XMMRegisterID dest)
+    {
+        switch (lane) {
+        case SIMDLane::f32x4:
+            m_formatter.vexNdsLigWigCommutativeTwoByteOp((OneByteOpcodeID)0, OP2_SUBPS, (RegisterID)dest, (RegisterID)left, (RegisterID)right);
+            break;
+        case SIMDLane::f64x2:
+            m_formatter.vexNdsLigWigCommutativeTwoByteOp(PRE_OPERAND_SIZE, OP2_SUBPD, (RegisterID)dest, (RegisterID)left, (RegisterID)right);
+            break;
+        default:
+            ASSERT_NOT_REACHED("Invalid SIMD lane for floating-point subtract.");
+        }
+    }
+
+    void vectorSub(SIMDLane lane, XMMRegisterID left, XMMRegisterID right, XMMRegisterID dest)
+    {
+        switch (lane) {
+        case SIMDLane::i8x16:
+            m_formatter.vexNdsLigWigTwoByteOp(PRE_OPERAND_SIZE, OP2_PSUBB, (RegisterID)dest, (RegisterID)left, (RegisterID)right);
+            break;
+        case SIMDLane::i16x8:
+            m_formatter.vexNdsLigWigTwoByteOp(PRE_OPERAND_SIZE, OP2_PSUBW, (RegisterID)dest, (RegisterID)left, (RegisterID)right);
+            break;
+        case SIMDLane::i32x4:
+            m_formatter.vexNdsLigWigTwoByteOp(PRE_OPERAND_SIZE, OP2_PSUBD, (RegisterID)dest, (RegisterID)left, (RegisterID)right);
+            break;
+        case SIMDLane::i64x2:
+            m_formatter.vexNdsLigWigTwoByteOp(PRE_OPERAND_SIZE, OP2_PSUBQ, (RegisterID)dest, (RegisterID)left, (RegisterID)right);
+            break;
+        default:
+            ASSERT_NOT_REACHED("Invalid SIMD lane for integer subtract.");
+        }
+    }
+
+    void vectorFmul(SIMDLane lane, XMMRegisterID left, XMMRegisterID right, XMMRegisterID dest)
+    {
+        switch (lane) {
+        case SIMDLane::f32x4:
+            m_formatter.vexNdsLigWigCommutativeTwoByteOp((OneByteOpcodeID)0, OP2_MULPS, (RegisterID)dest, (RegisterID)left, (RegisterID)right);
+            break;
+        case SIMDLane::f64x2:
+            m_formatter.vexNdsLigWigCommutativeTwoByteOp(PRE_OPERAND_SIZE, OP2_MULPD, (RegisterID)dest, (RegisterID)left, (RegisterID)right);
+            break;
+        default:
+            ASSERT_NOT_REACHED("Invalid SIMD lane for floating-point multiply.");
+        }
+    }
+
+    void vectorMul(SIMDLane lane, XMMRegisterID left, XMMRegisterID right, XMMRegisterID dest)
+    {
+        switch (lane) {
+        case SIMDLane::i16x8:
+            m_formatter.vexNdsLigWigCommutativeTwoByteOp(PRE_OPERAND_SIZE, OP2_PMULLW, (RegisterID)dest, (RegisterID)left, (RegisterID)right);
+            break;
+        case SIMDLane::i32x4:
+            m_formatter.vexNdsLigWigThreeByteOp(PRE_OPERAND_SIZE, VexImpliedBytes::ThreeBytesOp38, OP3_PMULLD, (RegisterID)dest, (RegisterID)left, (RegisterID)right);
+            break;
+        case SIMDLane::i64x2:
+            ASSERT_NOT_REACHED("TODO: Implement i64x2.mul. It's not natively supported on Intel so we need to do some weird things.");
```

Comment:
release assert not reached, and mention that it should have been lowered already

Reply:
I've switched the asserts to release asserts, and added a more informative message in the latest version.